cpp/arcticdb/pipeline/frame_utils.cpp

```diff
@@ -170,12 +170,33 @@ void adjust_slice_rowcounts(const std::shared_ptr<pipelines::PipelineContext>& p
 
 size_t adjust_slice_rowcounts(std::vector<pipelines::SliceAndKey> & slice_and_keys) {
     using namespace arcticdb::pipelines;
-    auto current_col = slice_and_keys[0].slice_.col_range.first;
+    auto current_col = slice_and_keys[0].slice_.col_range;
```

Comment:
Doing this by columns is quite complicated. Would it not be easier just to sort by row range first: 
```suggestion
   size_t adjust_slice_rowcounts(std::vector<pipelines::SliceAndKey> & slice_and_keys) {
	if(slice_and_keys.empty())
		return;

	std::sort(std::begin(slice_and_keys), std::end(slice_and_keys), [] (const auto& left, const auto& right) {
		return std::tie(left.slice.row_range, left.slice.col_range) < std::tie(right.slice.row_range, right.slice.col_range);
	}
	
	auto offset = slice_and_keys[0].slice_.row_range.first;
	auto diff = slice_and_keys[0].slice_.row_range.diff();
	
	for(auto& slice_and_key : slice_and_keys) {
		if(slice_and_key.row_range.first != offset) {
			offset += diff;
			diff = slice_and_key.slice_.row_range.diff();
		}
		slice_and_key.row_range = RowRange{offset, diff};
	}
	
	return offset + diff;
}
```

Reply:
Once we standardize the index sorting on rows-first, we can even drop the sort